### PR TITLE
Fix data file validation errors

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#75
ebi-ait/dcp-ingest-central#105

### CODE REVIEW

File: src/utils/ingest-client/ingest-file-validator.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ class IngestFileValidator {
      * @param fileFormat
      * @param fileName
      */
-    validateFile(fileResource: any, fileFormat: string, fileName: string) : Promise<ValidationJob> {
+    validateFile(fileResource: any, fileFormat: string, fileName: string): Promise<ValidationJob> {
```
Comment (jacobwindsor) [inline src/utils/ingest-client/ingest-file-validator.ts:34]: async function might make this a lot easier to reason about. Out of scope here, just something for the future.

**Review by jacobwindsor — APPROVED** [review]

File: src/utils/ingest-client/ingest-file-validator.ts (hunk 1 above)
Comment (prabh-t) [reply src/utils/ingest-client/ingest-file-validator.ts:34]: the validator codebase needs an overhaul to use the more recent and comfortable async/await syntax. I think it plays well with bluebird promises.

File: src/listener/handlers/document-update-handler.ts [hunk 2]
```diff
@@ -107,37 +136,49 @@ class DocumentUpdateHandler implements IHandler {
...
+                    // otherwise set validation state to INVALID, return error report with NoCloudUrl
+                    const msg = "File cloudUrl property not set.";
+                    const err = new ErrorReport(ErrorType.FileNotUploaded, msg);
+                    err.userFriendlyMessage = "File not uploaded.";
+
+                    contentValidationReport.validationState = "INVALID";
+
+                    if (!contentValidationReport.validationErrors || contentValidationReport.validationErrors.length == 0) {
+                        contentValidationReport.validationErrors = [];
+                    }
```
Comment (prabh-t) [inline src/listener/handlers/document-update-handler.ts:172]: is this really needed when we were previously setting (overriding) an validation errors anyway using
`new ValidationReport("INVALID", [err])`

Comment (aaclan-ebi) [reply src/listener/handlers/document-update-handler.ts:172]: ah this check is to just make sure contentValidationReport.validationErrors is not undefined when i do 
```
contentValidationReport.validationErrors.push(err)
```
actually i might not need the second condition

I need to append the "file not uploaded" error to the metadata errors.

**Pull request merged**